app/app-insights/app-insights.js

```diff
@@ -1,12 +1,16 @@
 const config = require('config');
 const appInsights = require('applicationinsights');
 
+const enabled = config.get('appInsights.enabled');
+
 const enableAppInsights = () => {
-  const appInsightsKey = config.get('appInsights.instrumentationKey');
-  appInsights.setup(appInsightsKey)
-    .setAutoDependencyCorrelation(true)
-    .setAutoCollectConsole(true, true)
-    .start();
+  if (enabled) {
```

Comment:
added a flag because of this error in the tests:

```
TypeError [ERR_INVALID_ARG_TYPE]: The "url" argument must be of type string. Received type undefined
    at Url.parse (url.js:106:11)
    at Object.urlParse [as parse] (url.js:100:13)
    at HttpDependencyParser.getDependencyTelemetry (node_modules/applicationinsights/out/AutoCollection/HttpDependencyParser.js:51:29)
    at OverriddenClientRequest.<anonymous> (node_modules/applicationinsights/out/AutoCollection/HttpDependencies.js:142:57)
    at ZoneDelegate.invokeTask (node_modules/zone.js/dist/zone-node.js:275:35)
    at Zone.runTask (node_modules/zone.js/dist/zone-node.js:151:47)
    at OverriddenClientRequest.ZoneTask.invoke (node_modules/zone.js/dist/zone-node.js:345:33)
    at _respond (node_modules/nock/lib/request_overrider.js:523:17)
    at respond (node_modules/nock/lib/request_overrider.js:506:11)
    at ZoneDelegate.invokeTask (node_modules/zone.js/dist/zone-node.js:275:35)
    at Zone.runTask (node_modules/zone.js/dist/zone-node.js:151:47)
    at ZoneTask.invoke (node_modules/zone.js/dist/zone-node.js:345:33)
    at args.(anonymous function) (node_modules/zone.js/dist/zone-node.js:1497:25)
    at process._tickCallback (internal/process/next_tick.js:150:11)
```

nock doesn't have the same properties as request, it doesn't provide a requestOptions or a method field so app insights was blowing up, tried a few things but gave up